pkg/services/dashboards/service/dashboard_service.go

```diff
@@ -1286,6 +1336,10 @@ func (dr *DashboardServiceImpl) setDefaultPermissions(ctx context.Context, dto *
 }
 
 func (dr *DashboardServiceImpl) setDefaultFolderPermissions(ctx context.Context, cmd *folder.CreateFolderCommand, f *folder.Folder, provisioned bool) {
+	if dr.features.IsEnabledGlobally(featuremgmt.FlagKubernetesClientDashboardsFolders) {
```

Comment:
Disabling setting permissions in the service when groing through k8s

Reply:
why?  ideally the frontend should start using k8s directly "soon" -- and this flag will be enabled by default

Reply:
I am expecting `SetDefaultPermissions` to be called when going through k8s.

Reply:
But clearly I am missing something in mode0.

Reply:
I'm currently in the process of fixing some permission issues that we have when going through /apis when not hitting unified storage and it involves calling this function here.

Are we sure checking the featureflag is enough? Shouldn't we rather be making this dependent on whether unified storage is currently being used?

I use `isReadingLegacy := dualwrite.IsReadingLegacyDashboardsAndFolders(ctx, dualWriter)` for that check.

Reply:
Thanks marco. I agree that in mode3, we might have issues if we read from one store but write to both. Will dig more into it.